### ISSUE TITLE
Account failed to Sync

### ISSUE THREAD

**Issue by Chluz** [operasoftware/ssh-key-authority#22]
Hi, 
just trying this nice tool out. 
I've been able to setup everything properly, but I have the following error when I launch a sync: 1 account failed to sync .
When I login via ssh as keys-sync to the server I am trying to sync to, and navigate to the /var/local/keys-sync folder I can create a file with no issue. 

Here is my log 
Sync process spawning for: 2/
Sep 30 21:39:24 piglet syncd.php[5483]: 2018-09-30T19:39:21+00:00 linux-server.dandc.luzzato.com: Preparing sync.
Sep 30 21:39:24 piglet syncd.php[5483]: 2018-09-30T19:39:21+00:00 linux-server.dandc.luzzato.com: Checking IP address 192.168.48.7.
Sep 30 21:39:24 piglet syncd.php[5483]: 2018-09-30T19:39:21+00:00 linux-server.dandc.luzzato.com: Attempting to connect.
Sep 30 21:39:24 piglet syncd.php[5483]: 2018-09-30T19:39:22+00:00 linux-server.dandc.luzzato.com: Logged in as keys-sync.
Sep 30 21:39:24 piglet syncd.php[5483]: linux-server.dandc.luzzato.com: Sync command execution failed for charles-luzzato, file_put_contents(ssh2.sftp://Resource id #31/var/local/keys-sync/charles-luzzato): failed to open stream: operation failed
Sep 30 21:39:24 piglet syncd.php[5483]: 2018-09-30T19:39:23+00:00 linux-server.dandc.luzzato.com: Sync finished

Any idea welcome, and thanks again for this nice tool !

**Comment by thomas-pike** [comment]
This is certainly very strange. The first thing I usually suggest in these cases is something that you have already tried (login to the account via SSH and try creating a file in that directory manually).

Since you have already tried this, I don't have much to suggest, but:

* Checking the ssh/auth logs on the target server for obvious errors would be the place I'd start.
* If the `/var/local/keys-sync/charles-luzzato` file exists, check that it is writable by the keys-sync user (doesn't have weird permissions or attributes).

**Comment by Chluz** [comment]
Hi Thomas, 

I found some time to figure this out. It looks like php forgot how to read its own datasets, in newer versions at least, as I am running PHP 7.0.32-0ubuntu0.16.04.1 
The problem was caused by the bug listed here https://bugs.php.net/bug.php?id=73597 .
I fixed it by changing line 413 from 

```
$remote_entity = "ssh2.sftp://$sftp$remote_filename";
```
to 
```
$remote_entity = "ssh2.sftp://" . intval($sftp) . $remote_filename;
```

I have not tested this extensively to see if it needs to be patched anywhere else. Would you like me to make a branch pull request ?
Kind regards

**Comment by kujoukaren318** [comment]
This fix my problem, thanks Chluz. I think Thomas should change this to avoid this problem. 
This is a nice tool by the way.

**Comment by thomas-pike** [comment]
Thanks @Chluz for taking the time to investigate this. This seems like it should be a relatively safe change to make, please go ahead with a pull request.

Best regards
Thomas

**Comment by thomas-pike** [comment]
Resolved in #26.
